src/main/scala/com/fulcrumgenomics/bam/SamRecordClipper.scala

```diff
@@ -48,6 +48,13 @@ import scala.collection.mutable.ListBuffer
 trait SamRecordClipper {
   import SamRecordClipper._
 
+  /**
+    * Should extended attributes that are the same length as the bases and qualities
+    * be automatically clipped in the same way as the bases and qualities.
+    * @return
+    */
+  def autoClipAttributes: Boolean
```

Comment:
Looking at a few of the tools, it looks like `mode` is always either hard-clip or soft-clip, so why not the same treatment?